raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -325,8 +341,30 @@ private void fireHandleResign() {
     }
 
     @Override
-    public void initialize() throws IOException {
-        quorum.initialize(new OffsetAndEpoch(log.endOffset().offset, log.lastFetchedEpoch()));
+    public void initialize(String quorumVoterStrings) throws IOException {
```

Comment:
This is a little awkward given the presence of the voter configuration in `RaftConfig`. Perhaps we could let `initialize` take the full config?
```java
public void initialize(RaftConfig config) throws IOException
```

Reply:
How about storing the `RaftConfig` as a field of `KafkaRaftClient`? This would allow us to remove a few of the configuration parameter passed through the constructor.

@hachikuji is there a reason why `KafkaRaftClient` has one constructor which accepts a `RaftConfig` and one constructor that accepts the configurations as explicit parameters?

Reply:
@jsancio Not really. The one constructor is used in test cases. However, the tests could just as well build a `RaftConfig`, which is what I assume we would have to do to implement my suggestion above.

> How about storing the RaftConfig as a field of KafkaRaftClient? This would allow us to remove a few of the configuration parameter passed through the constructor.

I debated suggesting that as well. I decided not to because it seemed vexing to do a `Map.get` and parse every time we need the fetch timeout... On the other hand, we could move the fields into `RaftConfig`.

Reply:
I had the same idea as @jsancio. I think it makes sense to store a reference to RaftConfig locally and access these parameters on-demand. And caching these in private fields on construction of the RaftConfig makes sense to me. That would prevent the `Map.get` on every access.